rotkehlchen/tasks/assets.py

```diff
@@ -192,74 +198,88 @@ def augmented_spam_detection(user_db: DBHandler) -> None:
     with (
         user_db.conn.read_ctx() as cursor,
         globaldb.conn.read_ctx() as globaldb_cursor,
     ):
-        cursor.execute('SELECT DISTINCT asset FROM history_events')
-        assets = [row[0] for row in cursor]
         # query assets marked as false positive
         false_positive_ids = set(globaldb_get_general_cache_values(
             cursor=globaldb_cursor,
             key_parts=(CacheType.SPAM_ASSET_FALSE_POSITIVE,),
         )) | KNOWN_FALSE_POSITIVES
 
-        globaldb_cursor.execute(  # take only assets that are in the global DB
-            f'SELECT identifier FROM assets WHERE identifier IN ({", ".join(["?"] * len(assets))}) AND type=?',  # noqa: E501
-            [*assets, AssetType.EVM_TOKEN.serialize_for_db()],
+        # get transactions with a single event which is of type receive/none
+        cursor.execute(
+            """
+            SELECT he.asset, "0x" || hex(eei.tx_hash)
+            FROM history_events he
+            JOIN evm_events_info eei ON he.identifier = eei.identifier
+            WHERE he.type = ?
+            AND he.subtype = ?
+            AND he.event_identifier IN (
+                SELECT event_identifier
+                FROM history_events
+                GROUP BY event_identifier
+                HAVING COUNT(*) = 1
+            )
+            ORDER BY he.asset;
+            """,
+            (
+                HistoryEventType.RECEIVE.serialize(),
+                HistoryEventSubType.NONE.serialize(),
+            ),
         )
 
-        for (asset_id,) in globaldb_cursor:
-            if asset_id in false_positive_ids:
+        for asset_id, tx_hash in cursor:
+            if (
+                asset_id in false_positive_ids or
+                asset_id in processed_assets
+            ):
                 continue
 
-            asset = Asset(asset_id)
-            events: list[EvmEvent] = history_db.get_history_events(
-                cursor=cursor,
-                filter_query=EvmEventFilterQuery.make(
-                    assets=(asset,),
-                    order_by_rules=[('timestamp', True)],
-                    excluded_locations=list(EVMLIKE_LOCATIONS),  # exclude evm like events because they don't have EVM transactions  # noqa: E501
+            try:
+                token = EvmToken(asset_id)
+            except (UnknownAsset, WrongAssetType):
+                processed_assets.add(asset_id)
+                continue
+
+            # count the number of transfers in the receipt emitted by the contract
+            cursor.execute(
+                """
+                SELECT COUNT(*) FROM evmtx_receipts
+                JOIN evmtx_receipt_logs l ON evmtx_receipts.tx_id = l.tx_id
+                JOIN evmtx_receipt_log_topics t ON l.identifier = t.log
+                JOIn evm_transactions ON evm_transactions.identifier=evmtx_receipts.tx_id
+                WHERE tx_hash = ?
+                AND t.topic_index = 0  -- The event signature is always the first topic
```

Comment:
easier to understand comment (c - style)

```suggestion
                AND t.topic_index = 0  /* The event signature is always the first topic */
```